tests/test_interface.py

```diff
@@ -0,0 +1,91 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This file is part of ansible-nmstate.
+#
+# ansible-nmstate is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ansible-nmstate is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ansible-nmstate.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+""" unit tests for nmstate_linkagg.py """
+
+try:
+    from unittest import mock
+except ImportError:  # py2
+    import mock
+
+import sys
+
+sys.modules["ansible"] = mock.Mock()
```

Comment:
The problem is that the ansible module imports will not work, for example `from ansible.module_utils.ansible_nmstate import AnsibleNMState` does only work when running in he Ansible runtime environment but not in the pytest environment. The `ansible_nmstate` module is in the `module_utils` directory in the git repo.

I did not find a better way to mock/redirect these imports other than the current approach. Do you have a better idea?

Reply:
I meant this:
- Assuming production.py is tested and it has dependency on ansible.module_utils.ansible_nmstate
- To mock ansible_nmstate, I need to mock ansible at the usage:
   mock.patch('production.ansible'...)

But maybe I am missing something.

Reply:
production.py has this code:

```
import ansible.module_utils.ansible_nmstate
```

This does not work in the test environment because there is no such module. Ansible will copy the module from `module_utils` on live runs. But in the repository there is only `module_utils.ansible_nmstate`. AFAIU, to be able to mock something in production.py, I need to be able to import it first. Do you see a better way?

Reply:
> I need to be able to import it first.

I think you should be able to mock it prior importing it in the test (in conftest.py with autouse).
I will try it out this evening.

Reply:
I see, this is working. Fixed it.